test/performance/target/buffer.cpp

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Copyright 2024 Joel E. Anderson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmark/benchmark.h>
+#include <stumpless.h>
+#include "test/helper/memory_counter.hpp"
+#define MAX_BUFFER_SIZE (1024 * 1024 * 1024 * 1) // 1 GB
```

Comment:
Drop this by at least an order of magnitude so that it can run on smaller systems. Perhaps 4KiB?

Reply:
Hello, I'm sorry for the delay: I've got busy with personal stuff.

I've made a recent commit with the suggestion of setting this macro according to the user's cpu architecture. If you like this way, we can change the define into other magnitude values. Otherwise, we can rollback to defining it to 4KiB and keep it this way.

Reply:
I think for just a test, just stick with a single hard-coded value; it's probably not worth trying to optimize for different architectures since this is just a test.

Reply:
Nice, ok.
I've left as was suggested before: setting MAX_BUFFER_SIZE as 4KiB.

I hope to contribute more here in the future!

Reply:
I will add that if you are interested in architecture-specific improvements don't let this discourage you: there is definitely a place for this in the library code base. The page size is one example of a place for improvement: if it can't be queried from the system using something like `sysconf` then it defaults to a single value: if you can find some better fallback values for `FALLBACK_PAGESIZE` in `CMakeLists.txt` for different architectures this could be a separate contribution.